front50-proto/src/main/proto/applications.proto

```diff
@@ -0,0 +1,241 @@
+  // core data model.
+  map<string, AttributeGroup> attribute_groups = 9;
+
+  // A named collection of instance links for an application.
+  repeated InstanceLinkGroup instance_links = 10;
+
+  // A list of all accounts an application resides in.
+  //
+  // While not required, defining accounts will help inform Spinnaker's behavior for the application.
+  repeated string accounts = 11;
+
+  // A list of application-wide notification configurations.
+  //
+  // Notifications can be triggered in response to a number of events within Spinnaker, including pipeline execution
+  // lifecycle changes.
+  repeated Notification notifications = 12;
+
+  // An arbitrary bucket of labels for the application that can be used by end-users and integrators alike.
+  map<string, string> labels = 13;
+}
+
+// AttributeGroup is a named, client-typed bucket of data used for extending Front50's data model.
+message AttributeGroup {
+  // Resource name for the attribute group (e.g. "attributeGroups/netflixTimelines")
+  string name = 1;
+
+  // The data associated with the attribute group.
+  google.protobuf.Struct data = 2;
+
+  // The required permissions for updating an attribute group.
+  repeated Permission permissions = 3;
+}
+
+// InstanceLinkGroup is a titled collection of InstanceLinks.
+message InstanceLinkGroup {
+  // The title of the group.
+  string title = 1;
+
+  // A collection of instance links.
+  repeated InstanceLink links = 2;
+}
+
+// InstanceLink is a titled URL link template for server group instances.
+message InstanceLink {
+  // The title of an instance link.
+  string title = 1;
+
+  // The link template value.
+  string link = 2;
+}
+
+// Notification is the configuration for application-wide notification recipients.
+message Notification {
+  // The type of notification (e.g. slack, email, webhook...)
+  string notification_type = 1;
+
+  // The type-specific notification recipient (e.g. email address, slack channel(s), webhook endpoint...)
+  string recipient = 2;
+
+  // A collection of event names to notify on.
+  repeated string notifying_events = 3;
+
+  // A bucket of type-specific notification config.
+  map<string, string> metadata = 6;
+}
+
+// Permission defines an authorization and its allowed prinicpals.
+message Permission {
+  // The name of the authorization (e.g. WRITE, READ, EXECUTE...)
+  string name = 1;
+
+  // A collection of principals who are allowed this permission.
+  repeated string allowed_principals = 2;
+}
+
+// ListApplicationsRequest can be used to list and search applications known to Spinnaker.
+message ListApplicationsRequest {
+  // The number of applications to return in the response.
+  //
+  // Defaults to 100.
+  int32 page_size = 1;
+
+  // The page token to continue pagination from. If left unset, the first `page_size` results will be returned.
+  string page_token = 2;
+
+  // Allows filtering of applications by fields.
+  string filter = 3;
```

Comment:
What's an example of what this filter would look like?

Reply:
Hmm, I was 99% sure I saw this format defined in the Google Cloud API design doc, but now that I'm searching, I can't seem to find it. :\ From memory, it was something like `myField=foo;myField2=Blah`. Does this ring any bells for you Google folks?

Reply:
Most of the compute APIs have a similar field, e.g.:
https://cloud.google.com/compute/docs/reference/rest/v1/regionInstanceGroupManagers/list

But I don't think it's a standard syntax for which we have an open source parser, so you'll probably have to write your own...

Reply:
I already came to terms with writing a parser. Imagine we applied the same `filter` field & parser across all of our APIs where filtering functionality was required - how would you feel about this?

Reply:
Yeah, sure, seems reasonable.